### DISCUSSION TITLE
->formatStateUsing() used on relationship field

### DISCUSSION THREAD

**Discussion by Nestoran** [filamentphp/filament#7269] [Help]
->formatStateUsing() applies concatenation multiple times, when used inside a relationship.

```php
Forms\Components\Fieldset::make(trans('user.user_data'))
    ->relationship('data')
    ->schema([
        TextInput::make('country')
            ->label(trans('user.country'))
            ->formatStateUsing(fn (string $state): string => "countries.{$state}"),
    ])
```

![image](https://github.com/filamentphp/filament/assets/66977657/a73d68eb-dbbc-4582-ba53-15b612949230)

Work as expected if not used with `->relationship()`

![image](https://github.com/filamentphp/filament/assets/66977657/5364b9a0-ded4-4ad7-bb4d-6b2fd94b4a72)

Is there anything I should do but missed from the docs?

**Comment by danharrin** [comment]
Try `->formatStateUsing(fn (string $state): string => Str::start($state, 'countries.'))`

**Reply by Nestoran** [reply]
This works. But actually what I need would be this:
`->formatStateUsing(fn (string $state): string => Str::wrap($state, 'countries.', '.name')`

But this returns this again:
`countries.countries.348.name.name`

This works again, but not very nice:
`->formatStateUsing(fn (string $state): string => Str::finish(Str::start($state, 'countries.'), '.name')`

However when I try to use this to return a config value:
`->formatStateUsing(fn (string $state): string => config(Str::finish(Str::start($state, 'countries.'), '.name'))`
This fails to find the value in the config file.

While manually giving the string:
`->formatStateUsing(fn (string $state): string => config('countries.348.name')`
 finds the value, and works well.

**Reply by danharrin** [reply]
I admit it is a bit weird, maybe open an issue and report it as a bug

**Reply by Nestoran** [reply]
Hm.. I tried to recreate it, but it worked well in a fresh project. Do you have any tips maybe, where I should be looking to solve this issue?

**Reply by danharrin** [reply]
Try removing stuff from your current project until you figure out what causes it
